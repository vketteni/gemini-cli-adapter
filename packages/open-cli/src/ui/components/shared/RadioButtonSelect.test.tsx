/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import { render } from 'ink-testing-library';
import {
  RadioButtonSelect,
  type RadioSelectItem,
} from './RadioButtonSelect.js';
import { describe, it, expect } from 'vitest';

const ITEMS: Array<RadioSelectItem<string>> = [
  { label: 'Option 1', value: 'one' },
  { label: 'Option 2', value: 'two' },
  { label: 'Option 3', value: 'three', disabled: true },
];

describe('<RadioButtonSelect />', () => {
  it('renders a list of items and matches snapshot', () => {
    const { lastFrame } = render(
      <RadioButtonSelect items={ITEMS} onSelect={() => {}} isFocused={true} />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders with the second item selected and matches snapshot', () => {
    const { lastFrame } = render(
      <RadioButtonSelect
        items={ITEMS}
        initialIndex={1}
        onSelect={() => {}}
        isFocused={true}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders with numbers hidden and matches snapshot', () => {
    const { lastFrame } = render(
      <RadioButtonSelect
        items={ITEMS}
        onSelect={() => {}}
        isFocused={true}
        showNumbers={false}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders with scroll arrows and matches snapshot', () => {
    const manyItems = Array.from({ length: 20 }, (_, i) => ({
      label: `Item ${i + 1}`,
      value: `item-${i + 1}`,
    }));
    const { lastFrame } = render(
      <RadioButtonSelect
        items={manyItems}
        onSelect={() => {}}
        isFocused={true}
        showScrollArrows={true}
        maxItemsToShow={5}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders with special theme display and matches snapshot', () => {
    const themeItems: Array<RadioSelectItem<string>> = [
      {
        label: 'Theme A (Light)',
        value: 'a-light',
        themeNameDisplay: 'Theme A',
        themeTypeDisplay: '(Light)',
      },
      {
        label: 'Theme B (Dark)',
        value: 'b-dark',
        themeNameDisplay: 'Theme B',
        themeTypeDisplay: '(Dark)',
      },
    ];
    const { lastFrame } = render(
      <RadioButtonSelect
        items={themeItems}
        onSelect={() => {}}
        isFocused={true}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders a list with >10 items and matches snapshot', () => {
    const manyItems = Array.from({ length: 12 }, (_, i) => ({
      label: `Item ${i + 1}`,
      value: `item-${i + 1}`,
    }));
    const { lastFrame } = render(
      <RadioButtonSelect
        items={manyItems}
        onSelect={() => {}}
        isFocused={true}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders nothing when no items are provided', () => {
    const { lastFrame } = render(
      <RadioButtonSelect items={[]} onSelect={() => {}} isFocused={true} />,
    );
    expect(lastFrame()).toBe('');
  });
});
