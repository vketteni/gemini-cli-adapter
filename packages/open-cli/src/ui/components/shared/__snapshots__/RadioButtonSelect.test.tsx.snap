// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RadioButtonSelect /> > renders a list of items and matches snapshot 1`] = `
"● 1. Option 1
  2. Option 2
  3. Option 3"
`;

exports[`<RadioButtonSelect /> > renders a list with >10 items and matches snapshot 1`] = `
"●  1. Item 1
   2. Item 2
   3. Item 3
   4. Item 4
   5. Item 5
   6. Item 6
   7. Item 7
   8. Item 8
   9. Item 9
  10. Item 10"
`;

exports[`<RadioButtonSelect /> > renders with numbers hidden and matches snapshot 1`] = `
"● 1. Option 1
  2. Option 2
  3. Option 3"
`;

exports[`<RadioButtonSelect /> > renders with scroll arrows and matches snapshot 1`] = `
"▲
●  1. Item 1
   2. Item 2
   3. Item 3
   4. Item 4
   5. Item 5
▼"
`;

exports[`<RadioButtonSelect /> > renders with special theme display and matches snapshot 1`] = `
"● 1. Theme A (Light)
  2. Theme B (Dark)"
`;

exports[`<RadioButtonSelect /> > renders with the second item selected and matches snapshot 1`] = `
"  1. Option 1
● 2. Option 2
  3. Option 3"
`;
